server/cmwell-ws/app/security/EagerAuthCache.scala

```
@@ -0,0 +1,125 @@
+        logger.info(s"AuthCache(nbg=$nbg) user $userName was not in memory, but added to Map")
+        if(nbg)
+          nData = nData.copy(users = nData.users + (userName -> user))
+        else
+          oData = oData.copy(users = oData.users + (userName -> user))
+        user
+      }
+    }
+  }
+
+  def invalidate(nbg: Boolean): Unit = load()
+
+  private def load(): Unit = {
+    retryUntil[AuthData](isSuccessful = !_.isEmpty, maxRetries = 30, delay = 10.seconds)(loadOnce(false)).map { data => oData = combiner(oData, data) }
+    delayedTask(5.seconds)(
+      retryUntil[AuthData](isSuccessful = !_.isEmpty, maxRetries = 30, delay = 10.seconds)(loadOnce(true)).map { data => nData = combiner(nData, data) }
+    )
+  }
+
+  private def loadOnce(nbg: Boolean): Future[AuthData] = {
+    // one level under /meta/auth is a parent (e.g. "users", "roles")
+    def isParent(infoton: Infoton) = infoton.path.count(_ == '/') < 4
+    logger.info(s"AuthCache(nbg=$nbg) is now loading...")
+    crudServiceFS.search(Some(PathFilter("/meta/auth", descendants = true)), withData = true, paginationParams = PaginationParams(0, 2048), nbg = nbg).map { searchResult =>
+      val data = searchResult.infotons.filterNot(isParent).map(i => i.path -> extractPayload(i)).collect { case (p, Some(jsv)) => p -> jsv }.toMap
+      val (usersData, rolesData) = cmwell.util.collections.partitionWith(data) { t =>
+        val (path, payload) = t
+        val isUser = path.startsWith("/meta/auth/users")
+        val key = path.substring(path.lastIndexOf("/") + 1)
+        if (isUser) Left(key -> payload)
+        else Right(key -> payload)
+      }
+      logger.info(s"AuthCache(nbg=$nbg) Loaded with ${usersData.size} users and ${rolesData.size} roles.")
+      AuthData(usersData.toMap, rolesData.toMap)
+    }.recover { case t: Throwable =>
+      logger.info(s"AuthCache(nbg=$nbg) failed to load because ${t.getMessage}")
```

don't hide the error. let's change the log to:
`logger.error(s"AuthCache(nbg=$nbg) failed to load", t)`